runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -73,12 +78,13 @@ public String getTransformId(AppliedPTransform<?, ?, ?> pTransform) {
   }
 
   /**
-   * Puts the provided {@link PCollection} into this {@link SdkComponents} if it is not already
-   * present, returning the id of the provided {@link PCollection} in this {@link SdkComponents}.
+   * Registers the provided {@link PCollection} into this {@link SdkComponents}, returning a unique
+   * ID for the {@link PCollection}. Multiple calls to {@link #registerPCollection(PCollection)}
+   * with the same {@link PCollection}will return the same unique ID.
    *
    * <p>Uses a unique name based on the result of {@link PCollection#getName()}.
```

Comment:
I would drop the bit about uses a unique name, since these are internal details. For all intensive purposes your returning random ids.